source/extensions/common/wasm/v8/v8.cc

```diff
@@ -241,15 +252,33 @@ template <typename T, typename U> constexpr T convertValTypesToArgsTuple(const U
 
 // V8 implementation.
 
-bool V8::load(const std::string& code, bool /* allow_precompiled */) {
+bool V8::load(const std::string& code, bool allow_precompiled) {
   ENVOY_LOG(trace, "load()");
   store_ = wasm::Store::make(engine());
   RELEASE_ASSERT(store_ != nullptr, "");
 
   source_ = wasm::vec<byte_t>::make_uninitialized(code.size());
   ::memcpy(source_.get(), code.data(), code.size());
 
-  module_ = wasm::Module::make(store_.get(), source_);
+  if (allow_precompiled) {
+    const auto precompiled = getCustomSection(getPrecompiledSectionName());
+    if (!precompiled.empty()) {
+      auto vec = wasm::vec<byte_t>::make_uninitialized(precompiled.size());
+      ::memcpy(vec.get(), precompiled.data(), precompiled.size());
+
+      module_ = wasm::Module::deserialize(store_.get(), vec);
+      if (!module_) {
+        // Precompiled module that cannot be loaded is considered a hard error,
```

Comment:
Log here.  In WAVM we throw a WasmVmException with a description of the problem.  We should standardize on how we want to deal with these sorts of failures.

Reply:
WAVM doesn't throw an exception in `load()` (which returns `true/false`, since we're a bit inconsistent with use of exceptions and `bool`).

Furthermore, it doesn't look that it validates that module was successfully loaded, but always returns `true`:
https://github.com/envoyproxy/envoy-wasm/blob/cc4af107e98755ae02e6b2b1e99eb49157c1793b/source/extensions/common/wasm/wavm/wavm.cc#L265-L270